addons/sale_timesheet/models/sale_order.py

```diff
@@ -146,6 +147,21 @@ def action_view_timesheet(self):
             action = {'type': 'ir.actions.act_window_close'}
         return action
 
+    def _create_invoices(self, grouped=False, final=False, date=None):
+        """ Override the _create_invoice method in sale.order model in sale module
+
+            Add new parameter in this method, to invoice sale.order with a date. This date is used in sale_make_invoice_advance_inv into this module.
+
+            :param date
+
+            :return {account.move}: the invoices created
+        """
+        moves = super(SaleOrder, self)._create_invoices(grouped, final)
+
+        moves.search_timesheet_to_link(date)
```

Comment:
This method doesn't only search records.

It also write on them. So find a better method name ;)